### ISSUE TITLE
Error: must provide image ID and image name to use an image: invalid argument

### ISSUE THREAD

**Issue by michielhemme** [containers/libpod#3345]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**

**Steps to reproduce the issue:**

1. Install Podman 1.3.2 on RHEL 7.7 with basic repo, extras and beta, and make a test user without root rights.

2. Login with the test user.

3. Try to run multiple containers.

**Describe the results you received:**
Second container gives error that i need to provide Image ID. 
[test@rhel77 ~]$ podman run -d nginx
25042d922c25eaf726518a87460b51d12b5ecc791b2e988349738832935c86c1
[test@rhel77 ~]$ podman run -d nginx
Error: must provide image ID and image name to use an image: invalid argument

**Describe the results you expected:**
That the second container would start.
[test@rhel77 ~]$ podman run -d nginx
25042d922c25eaf726518a87460b51d12b5ecc791b2e988349738832935c86c1
[test@rhel77 ~]$ podman run -d nginx
d12b5ecc791b2e988349738832935c86c125042d922c25eaf726518a87460b51

**Additional information you deem important (e.g. issue happens only occasionally):**
When I stop the only container active I can start one again.

**Output of `podman version`:**

```
podman version
Version:            1.3.2
RemoteAPI Version:  1
Go Version:         go1.10.8
OS/Arch:            linux/amd64
```

**Output of `podman info --debug`:**

```
podman info --debug
debug:
  compiler: gc
  git commit: ""
  go version: go1.10.8
  podman version: 1.3.2
host:
  BuildahVersion: 1.8.2
  Conmon:
    package: podman-1.3.2-1.git14fdcd0.el7.x86_64
    path: /usr/libexec/podman/conmon
    version: 'conmon version 1.14.0-dev, commit: 4780496c1f6949e21c807e17b57c183e9424c90a-dirty'
  Distribution:
    distribution: '"rhel"'
    version: "7.7"
  MemFree: 174981120
  MemTotal: 3973541888
  OCIRuntime:
    package: runc-1.0.0-83.dev.gitf56b4cb.fc28.x86_64
    path: /usr/bin/runc
    version: 'runc version spec: 1.0.1-dev'
  SwapFree: 3434860544
  SwapTotal: 3435130880
  arch: amd64
  cpus: 1
  hostname: rhel77.michiel
  kernel: 3.10.0-1049.el7.x86_64
  os: linux
  rootless: false
  uptime: 139h 46m 20.66s (Approximately 5.79 days)
registries:
  blocked: null
  insecure: null
  search:
  - registry.access.redhat.com
  - docker.io
  - registry.fedoraproject.org
  - quay.io
  - registry.centos.org
store:
  ConfigFile: /etc/containers/storage.conf
  ContainerStore:
    number: 18
  GraphDriverName: overlay
  GraphOptions: null
  GraphRoot: /var/lib/containers/storage
  GraphStatus:
    Backing Filesystem: xfs
    Native Overlay Diff: "true"
    Supports d_type: "true"
    Using metacopy: "false"
  ImageStore:
    number: 2
  RunRoot: /var/run/containers/storage
  VolumePath: /var/lib/containers/storage/volumes

```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

**Comment by rhatdan** [comment]
This is strange, I don't see an issue on Fedora.
If you attempt to run nginx a third time, does it continue to give you the same error?
Of course I am using a slightly newer version.
 ```
podman version
Version:            1.4.0
RemoteAPI Version:  1
Go Version:         go1.12.5
OS/Arch:            linux/amd64
```

**Comment by mheon** [comment]
Can't reproduce with 1.3.2, rootless, on F30. I'll try 7.7 later, but this one seems like it could be an environment issue.

**Comment by michielhemme** [comment]
@rhatdan I'm using podman 1.3.2, runc 1.0.0-rc8+dev and slirp4netns-0.3.0.2

[test@localhost ~]$ podman ps
CONTAINER ID  IMAGE                           COMMAND               CREATED         STATUS             PORTS  NAMES
5cf40bbb4d91  docker.io/library/nginx:latest  nginx -g daemon o...  23 seconds ago  Up 23 seconds ago         infallible_napier
[test@localhost ~]$ podman run -d nginx
Error: must provide image ID and image name to use an image: invalid argument
[test@localhost ~]$ podman stop 5cf4
5cf40bbb4d91f7ec92a66e61bda2c98f6c98ffbf2c8e68f628daa3f84a138950
[test@localhost ~]$ podman run -d nginx
7468b471f8995ce268748902feca34c4b606e913f14b0210f044a1b018076171
[test@localhost ~]$ podman run -d nginx
Error: must provide image ID and image name to use an image: invalid argument
[test@localhost ~]$ podman run -d nginx
Error: must provide image ID and image name to use an image: invalid argument
[test@localhost ~]$ podman run -d nginx
Error: must provide image ID and image name to use an image: invalid argument

@mheon Just installed it on another system. Same error occurs.

**Comment by mheon** [comment]
Alright, so it's consistent, but seemingly only on RHEL7

@michielhemme To confirm, are these fresh installs of Podman, or upgrading existing installations that have run Podman before?

**Comment by michielhemme** [comment]
@mheon, the podman installation is fresh out of the repositories. It is being delivered with runc-1.0.0-59-dev and trying to start a container then results in: 

[test@localhost ~]$ podman run -d nginx
Error: container create failed: cannot specify gid= mount options for unmapped gid in rootless containers
: internal libpod error

after upgrading runc to the latest version, I get the other error. That the first container can start but not the second one.

**Comment by rhatdan** [comment]
Could this be a vfs storage issue?  That is one difference.  RHEL7 does not support fuse-overlay.

**Comment by mheon** [comment]
Can't replicate with VFS on F30 - still seems RHEL specific

**Comment by rhatdan** [comment]
@giuseppe Any ideas?

**Comment by giuseppe** [comment]
the 7.7 kernel is missing some ioctl's we are using.  We will need to backport 4dca13e70423e657aec98999d9c6803075359517

**Comment by giuseppe** [comment]
it applies cleanly on top of v1.3.2, and it seems to solve the problem on RHEL 7.7

**Comment by mheon** [comment]
Is that in 1.4? We're expecting a 1.4 with 7.7, so if it's available there
we should be set

**Comment by giuseppe** [comment]
yes, it is included in 1.4

**Comment by rhatdan** [comment]
Closing since it is in Master and soon to be everywhere.
